static/app/views/app/index.tsx

```diff
@@ -35,201 +31,151 @@ const GlobalNotifications = HookOrDefault({
-  user?: Config['user'];
-  organization?: Organization;
+  children: React.ReactNode;
 };
 
-class App extends Component<Props, State> {
-  state: State = {
-    loading: false,
-    error: false,
-    needsUpgrade: ConfigStore.get('user')?.isSuperuser && ConfigStore.get('needsUpgrade'),
-    newsletterConsentPrompt: ConfigStore.get('user')?.flags?.newsletter_consent_prompt,
-  };
-
-  componentDidMount() {
-    this.props.api.request('/organizations/', {
-      query: {
-        member: '1',
-      },
-      success: data => {
-        OrganizationsStore.load(data);
-        this.setState({
-          loading: false,
-        });
-      },
-      error: () => {
-        this.setState({
-          loading: false,
-          error: true,
-        });
-      },
-    });
-
-    this.props.api.request('/internal/health/', {
-      success: data => {
-        if (data && data.problems) {
-          data.problems.forEach(problem => {
-            AlertActions.addAlert({
-              id: problem.id,
-              message: problem.message,
-              type: getAlertTypeForProblem(problem),
-              url: problem.url,
-            });
-          });
-        }
-      },
-      error: () => {}, // TODO: do something?
-    });
+/**
+ * App is the root level container for all routes.
+ */
+function App({config, children}: Props) {
+  const api = useApi();
+  const {organization} = useLegacyStore(OrganizationStore);
 
-    ConfigStore.get('messages').forEach(msg => {
-      AlertActions.addAlert({
-        message: msg.message,
-        type: msg.level,
-        neverExpire: true,
-      });
-    });
+  // Command palette global-shortcut
+  useHotkeys('command+shift+p, command+k, ctrl+shift+p, ctrl+k', e => {
+    openCommandPalette();
+    e.preventDefault();
+  });
+
+  // Theme toggle global shortcut
+  useHotkeys(
+    'command+shift+l, ctrl+shift+l',
+    e => {
+      ConfigStore.set('theme', config.theme === 'light' ? 'dark' : 'light');
+      e.preventDefault();
+    },
+    [config.theme]
+  );
 
-    if (DEPLOY_PREVIEW_CONFIG) {
-      displayDeployPreviewAlert();
-    } else if (EXPERIMENTAL_SPA) {
-      displayExperimentalSpaAlert();
+  /**
+   * Loads the users organization list into the OrganizationsStore
+   */
+  async function loadOrganizations() {
+    try {
+      const data = await api.requestPromise('/organizations/', {query: {member: '1'}});
+      OrganizationsStore.load(data);
+    } catch {
+      // TODO: do something?
     }
+  }
 
-    initApiClientErrorHandling();
+  /**
+   * Creates Alerts for any internal health problems
+   */
+  async function checkInternalHealth() {
+    let data: any = null;
 
-    const user = ConfigStore.get('user');
-    if (user) {
-      HookStore.get('analytics:init-user').map(cb => cb(user));
+    try {
+      data = await api.requestPromise('/internal/health/');
+    } catch {
+      // TODO: do something?
     }
 
-    fetchGuides();
-  }
+    data?.problems?.forEach?.(problem => {
```

Comment:
should we move it to inside the try-catch? because if an error happens the data will be null anyway. Inside the try-catch, we can also cast the data with a better type other than any

Reply:
I think I would rather the try/catch only catch API errors